### PR TITLE
Add information about A100 GPUs

### LINKED ISSUE #604: A100 release information
### On https://docs.rc.uab.edu/

- Add an announcement admonition about the A100 release (expires 60 days, make an issue w/ due date)

### On https://docs.rc.uab.edu/cheaha/slurm/gpu/#gpus

- Add an announcement admonition about the A100 release (expires 60 days, make an issue w/ due date)
- Add an FAQ section for the A100 release

### FAQ Section

- How do I access the A100 nodes?
    - Request jobs on the "amperenodes" partition (up to 12 hours) or "amperenodes-medium" partition (up to 48 hours).
- How many GPUs can I request at once?
    - Up to four GPUs may be requested by any one researcher at once.
    - There are two GPUs per node, so requesting four GPUs will allocate two nodes.
- What performance improvements can I expect over the P100 GPUs?
    - Performance improvements depend on the software and algorithms being used.
    - Swapping a single P100 to a single A100, you can generally expect 3x to 20x improvement.
    - ADD DOCUMENTATION FROM NVIDIA
- How can I make the most efficient use of the A100 GPUs?
    - A100s process data very rapidly. Ideally, we want the A100 to be the bottleneck during processing.
    - Possible ways to improve performance include...
        - ...copying your input data onto `/local/$SLURM_JOB_ID` (node-specific NVMe drives) before processing.
        - ...using a larger number of CPU cores for data loading and preprocessing.  `Need to qualify this statement, open up a dialog`
            - **NOTE FOR WRITER** This is becoming a concern as models and data increase is size and scope. We want to learn with you what configurations benefit most from additional resources
            - **NOTE FOR WRITER** This may need to go to docs
        - ...verifying improvements empirically by recording timings with different setups.
- Where are the A100 nodes physically located, and will this impact my workflows?
    - The A100 nodes are located in the DC BLOX Data Center, west of UAB Campus.
    - Because Cheaha storage (GPFS) is located on campus, there may be slightly higher latency when transferring data between the A100 nodes and GPFS. Impacts will only occur if very small amounts of data are transferred very frequently, which is unusual for most GPU workflows.
    - We strongly recommend copying your input data onto `/local/$SLURM_JOB_ID` prior to processing.
- What will happen to the P100 GPUs?
    - We intend to retain all of the 18 existing P100 GPU nodes.
    - 9 nodes are available now.
    - 9 nodes have been temporarily taken offline as we reconfigure hardware, and will be reallocated based on demand.
- What else should I be aware of?
    - Please be sure to clean your data off of `/local/$SLURM_JOB_ID` as soon as you no longer need it, before the job finishes. <point to example job>

### Example Job

Conceptually, it is possible to wrap existing `sbatch` script payloads with a `cp` command to move the data to the local SSD, then clean up the local SSD afterward. The specifics of the following example may need to be tweaked to meet the needs of the specific script.

```shell
#!/bin/bash
#SBATCH ...
#SBATCH --partition=amperenodes
#SBATCH --gres=gpu:1

# LOAD CUDA MODULES
module load CUDA/12.1.1
module load cuDNN/12.1.1

# CREATE TEMPORARY DIRECTORY
# WARNING! $TMPDIR will be deleted at the end of the script!
# Changing the following line can cause permanent, unintended deletion of important data.
TMPDIR="/local/$SLURM_JOB_ID"
mkdir -p "$TMPDIR"

# COPY RESEARCH DATA TO LOCAL TEMPORARY DIRECTORY
# Replace $MY_DATA_DIR with the path to your data folder
cp -r "$MY_DATA_DIR" "$TMPDIR"

# YOUR ORIGINAL WORKFLOW GOES HERE
...

# CLEAN UP TEMPORARY DIRECTORY
# WARNING!
# Changing the following line can cause permanent, unintended deletion of important data.
rm -rf "$TMPDIR"
```

The example above assumes one GPU on one node, a single processing step and no intermediate data being stored on GPFS. Any intermediate data would also need to be copied and cleaned up.

For workflows across multiple nodes, take care to ensure data is moved to local drives on the same node that the data will be processed on. Please contact us for more information if you need to use A100s on multiple nodes in a single job.

### LINKED ISSUE #605: Scrub `/scratch/local` and replace with `TMPDIR=/local/$SLURM_JOB_ID`
### What would you like to see added?

See title. We want to scrub `/scratch/local` (a symlink to `/local`) and replace it with `TMPDIR=/local/$SLURM_JOB_ID`, in the context of jobs. This resolves multiple potential issues.

- `/local` is used by A100 nodes for high-speed I/O to avoid I/O bottlenecks
- `$TMPDIR` is used by many software packages as a temporary directory
- Sets the stage for SLURM prolog/epilog automated temporary data cleanup

### LINKED ISSUE #606: CUDA module changes
### What would you like to see added?

Announce changes for CUDA libraries. All researchers should start using the following modules when using GPUs.

- `cudaX.Y/toolkit/X.Y.Z` is deprecated. We can no longer provide support for these modules. New modules of this form will not be added. Instead, we will be supporting `CUDA/X.Y.Z`
- Replace `cudaX.Y/toolkit/X.Y.Z` with `CUDA/X.Y.Z` whenever possible
- If working with deep learning models be sure to load `cuDNN/CUDA-X.Y.Z` with a matching version number.

### PR DESCRIPTION
**Pull request by wwarriner**
# Pull Request

## Overview

Adds info about A100 GPUs to multiple pages. Added info about new CUDA modules.

## Proposed Changes

- Adds information about using A100 GPUs effectively to gpu.md
- Adds information about IO Performance with `/local`
- Adds example script showing a template wrapper for IO performance
- Adds generalized example script to LOCAL_SCRATCH part of storage.md
- Adds `amperenodes` wherever `pascalnodes` is mentioned
- Adds announcement to main page
- Remove other announcement on hardware.md
- Modifies main page to reduce verbosity and make more compact
- Linting run on case_studies.md page. **No other changes made here, see #614**
- Simplified and clarified OOD Interactive Jupyter Apps environment setup field
- Added new CUDA and cuDNN module on Jupyter Apps
- Added GPU info to GPU page, extracted to table
- Added FAQ for A100 GPUs

## Related Issues

Fixes #604
Fixes #605 
Fixes #606

### CODE REVIEW

File: docs/education/case_studies.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ A license is no longer required to use Clara Parabricks 4.x and later versions,
 
 1. Access to internet.
 
-2. Any GPU that supports CUDA architecture 60, 61, 70, 75 and has 12GB GPU RAM or more. It has been tested on NVIDIA V100, NVIDIA A100, and NVIDIA T4 GPUs.
+2. Any GPU that supports CUDA architecture 6.0, 6.1, 7.0, 7.5 and has 12GB GPU RAM or more. It has been tested on NVIDIA V100, NVIDIA A100, and NVIDIA T4 GPUs.
```
Comment (Premas) [inline docs/education/case_studies.md:19]: Could you update the CUDA architecture, GPU RAM, and the test platform? 
I found the recent document has this update: https://docs.nvidia.com/clara/parabricks/4.1.0/gettingstarted.html.

Comment (wwarriner) [reply docs/education/case_studies.md:19]: Will do, thank you!

File: docs/cheaha/hardware.md [hunk 2]
```diff
@@ -20,13 +20,7 @@ Examples of how to make use of the table:
...
 - You submit 5 jobs to the "medium" partition and 5 to the "long" partition. It is possible that all 10 jobs may start, because partition node limits are separate. If all 5 jobs start, jobs on the "medium" partition.
-- You submit jobs to the "amperenodes" and "amperenodes-medium" partitions totaling 10 GPUs, and the "pascalnodes" partition totaling 4 GPUs. Jobs totaling 8 or fewer GPUs on the "gpu: ampere" group, and all jobs on "gpu: pascal", can start at the same time.
-
-<!-- markdownlint-disable MD046 -->
-!!! announcement
-
-    Physical A100 nodes are slated for a future release on Cheaha. Information related to `amperenodes` and the A100 GPUs should be considered tentative information. The `amperenodes` hardware is not yet available as of 2023-08-22. Release date to be determined (TBD).
-<!-- markdownlint-enable MD046 -->
+- You submit jobs to the "amperenodes" and "amperenodes-medium" partitions totaling 10 GPUs, and the "pascalnodes" partition totaling 4 GPUs. Jobs totaling 4 or fewer GPUs on the "gpu: ampere" group, and all jobs on "gpu: pascal", can start at the same time.
```
Comment (Premas) [inline docs/cheaha/hardware.md:23]: I felt this may require rewording. Could you please reword the sentence?

**Review by Premas — APPROVED** [review]
Everything looks good to me. I have added a few comments to the hardware.md and case_studies.md page. Please review them.

**Review by mdefende — DISMISSED** [review]

File: docs/education/case_studies.md (hunk 1 above)
Comment (wwarriner) [reply docs/education/case_studies.md:19]: From our discussion it sounds like they've dropped official support for P100s, but they do work with Parabricks 4.0.0. We'll want to test with 4.1.0 as well.

Applies to #614 as well.

Comment (Premas) [reply docs/education/case_studies.md:19]: Thanks for noting! 
The version tested was 4.0.0. Here is the link to documentation : https://docs.nvidia.com/clara/parabricks/4.0.0/gettingstarted.html. Thanks for notin

Comment (wwarriner) [reply docs/education/case_studies.md:19]: We're going to leave the docs at 4.0.0 for now, and update to 4.1.0 at a later time when we've had a chance to test.

File: docs/cheaha/hardware.md (hunk 2 above)
Comment (wwarriner) [reply docs/cheaha/hardware.md]: Taken care of in latest. We need to get this out, further suggestions are welcome and can be made into new issues.

**Comment by wwarriner** [comment]
Reviews were invalidated by my latest pushes. I'm going to self-approve (the horror!!) based on both y'alls previous approvals.

**Review by Premas — APPROVED** [review]

**Review by mdefende — APPROVED** [review]

**Pull request merged**